docs/development/AutomaticInstrumentation.md

```diff
@@ -276,3 +285,152 @@ There are some current limitations with what types/methods with our `CallTarget`
+
+- The Test should inherit from the abstract `TracingIntegrationTest` class.
+  - You will need to implement the `ValidateIntegrationSpan` method, which is used to validate spans have the required tags.
+- Pass your sample name in the base constructor
+- Start a mock agent in your integration tests
+- Use snapshot testing to assert that subsequent changes are correctly represented in traces
+
+##### Add required docker services
+
+If you need to run a database/message queue/other service to support your integration, you will need to update [the docker-compose.yml file](https://github.com/DataDog/dd-trace-dotnet/blob/master/docker-compose.yml)
+
+- Add the docker-compose service to the file.
+- Consider if you need a different service running on ARM64.
+- Make sure to "connect" the `IntegrationTests` and `IntegrationTests.Arm64` test services to your database service using `depends_on`.
+- Make sure to "connect" the `StartDependencies` (and `StartDependencies.Arm64`) services to your database service using `depends_on`, and update the `command` to make sure for your database service is listening on the correct port.
+- Use environment variables to pass the host information to the sample running in the integration tests container.
+
+Once all these steps are complete, you're ready to test
+
+#### Local testing using Nuke
+
+How you test your changes locally depends on several factors:
+
+- Do you need to do multi-api-version testing (normal for integrations)?
+- Do you need to run a docker service?
+- Are you on Windows/Linux/macOS
+
+If you need a docker service (i.e. you updated docker-compose.yml), its typically easiest to _not_ use docker-compose, and instead run a "standalone" version of the service. This is due to the way docker networking works. 
+
+For example, to run a MongoDB service locally, you can use something like this:
+
+```bash
+docker run --rm -p 27017:27017 mongo:4.0.9
+```
+
+Which exposes `localhost:27017` on your host. This is one the easiest ways to test on Windows. Things are trickier on macOS (Someone who uses it should update this!)
+
+##### On Windows
+
+On Windows, we _don't_ typically run multi-api-version tests. There is experimental support locally for this, but you may run into issues. To run against a _single_ version (the default version lised in the sample) you can use:
+
+```bash
+# Build the tracer with your changes
+# You need to run this every time you change your integration
+# But not if you're just changing a test/sample
+./tracer/build.ps1 BuildTracerHome -buildConfiguration Debug
+
+# Build and run the integration tests, building only
+# your sample, and running only your new tests
+# You can choose whichever framework is appropriate
+./tracer/build.ps1 BuildAndRunWindowsIntegrationTests -buildConfiguration Debug -framework net462 -Filter MyNewIntegrationTests -SampleName Samples.MyNewSample
+```
+
+##### On macOs
+
+TBC
```

Comment:
```suggestion
On MacOs, you won't be able to run all tests as some images aren't arm compatible (e.g. Kafka), some tests are meant to run on Windows only (using VM fixes that though). That said the process is quite the same as  for Windows:

```bash
# Build the tracer with your changes
# You need to run this every time you change your integration
# But not if you're just changing a test/sample
./tracer/build.sh BuildTracerHome -buildConfiguration Debug

# Build and run the integration tests, building only
# your sample, and running only your new tests
# You can choose whichever framework is appropriate
./tracer/build.sh BuildAndRunOSxIntegrationTests -buildConfiguration Debug -framework net6.0 -Filter MyNewIntegrationTests -SampleName Samples.MyNewSample
```
```